portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -1703,6 +1703,91 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 	    });
 	    return fetch_promise.promise();
 	}),
+	'getUnsequencedSamplesBasedOnSequencedSampleList': makeCachedPromiseFunction(
+		function(self, fetch_promise) {
+		    var sample_list_ids = self.getCancerStudyIds().map(function(cancer_study_id) {
+			return cancer_study_id + "_sequenced";
```

Comment:
Good point...In my opinion, using the stable id of case lists to drive business logic in the application should be avoided. However, this is spread by now in cBioPortal...for both case list stable ids and genetic profile stable ids. What @fedde-s suggests here is a good idea. I think this could be the first step for solving the issue on the longer run. Basically have a central place that can be the "place to query" when one has to find specific sample lists. For example, a utils file with a function like: 
```
function getSampleListIdsForDataTypes(dataTypes) {
    //business logic that returns one or more sample list ids based on given data types
    //...or perhaps just a wrapper here, with business logic implemented in backend
}
```
This business logic would still use the stable id matching to find the correct sample lists, but having it implemented in one place would make it easier to maintain or change it in the future. 

@jjgao @adamabeshouse what do you think of this idea?